parsec/core/fs/workspacefs/versioning_helpers.py

```diff
@@ -167,39 +246,39 @@ class ManifestDataAndPaths(NamedTuple):
     ):
         if early > late:
             return
-        manifest = await _load_manifest_or_cached(entry_id, version=version_number)
-        data = [
+        manifest = await manifest_cache.load(entry_id, version=version_number)
+        data = ManifestDataAndMutablePaths(
             ManifestData(
                 manifest.author,
                 manifest.updated,
                 is_folder_manifest(manifest),
                 None if not is_file_manifest(manifest) else manifest.size,
-            ),
-            None,  # Source path
-            None,  # Destination path
-            None,  # Current path
-        ]
+            )
+        )
         if len(target.parts) == path_level:
 
-            async def _populate_path_w_index(data, index, entry_id, timestamp):
-                try:
-                    data[index] = await _get_past_path(entry_id, timestamp=timestamp)
-                except (FSRemoteManifestNotFoundBadVersion, EntryNotFound):
-                    pass
+            async def _populate_source_path(data, entry_id, timestamp):
+                data.source = await _try_get_path_at_timestamp(entry_id, timestamp)
+
+            async def _populate_destination_path(data, entry_id, timestamp):
+                data.destination = await _try_get_path_at_timestamp(entry_id, timestamp)
+
+            async def _populate_current_path(data, entry_id, timestamp):
+                data.current = await _try_get_path_at_timestamp(entry_id, timestamp)
```

Comment:
So those attributes are only mutated once?

Reply:
Yes, I still use this class as a way to pass something that can be modified later by the populate_x_path methods, is it ok or is there a much cleaner way?